R/deploymentTarget.R

```diff
@@ -1,93 +1,241 @@
 # calculate the deployment target based on the passed parameters and
 # any saved deployments that we have
-deploymentTarget <- function(recordPath = ".",
-                             appId = NULL,
-                             appName = NULL,
-                             appTitle = NULL,
-                             envVars = NULL,
-                             account = NULL,
-                             server = NULL,
-                             forceUpdate = FALSE,
-                             error_call = caller_env()) {
+deploymentTarget <- function(
```

Comment:
Spurious WS change?

I wonder if we should change this name — it's currently a noun, which makes me think it should mostly be about creating an object; but actually it contains quite a lot of logic for finding the deployment. So maybe `findDeployment()`?

Reply:
Not a spurious whitespace change. I intentionally shifted to this style, but can revert if it's not a supported convention. With so many arguments and very long function names (e.g. the helpers called from here), the indentation felt a bit crazy.

We already have `findDeployment()`! 😜

Maybe `findDeploymentTarget()`, but I'll also try to explain why we need `findDeployment()` and the difference between the two "deployment discovery" functions.

Reply:
We generally reserve the style you moved to for cases where the style you moved from has arguments that don't all fit on one line. 

Hmmm, I had forgotten about `findDeployment()`. Maybe this function is something more like `findAndUpdateDeployment()`? Or is there some difference between a deployment and a "deployment target" that I've forgotten?

Reply:
This became `findDeployment()`